cloud/new_ec2.go

```diff
@@ -544,10 +522,14 @@ func (m *ec2Manager) CostForDuration(h *host.Host, start, end time.Time) (float6
 		return 0, errors.Wrap(err, "error creating client")
 	}
 
-	if m.onDemand(h) {
-		return m.costForDurationOnDemand(h, start, end)
-	} else if m.spot(h) {
-		return m.costForDurationSpot(h, start, end)
+	t := timeRange{start, end}
+	ec2Cost, err := pkgCachingPriceFetcher.getEC2Cost(m.client, h, t)
+	if err != nil {
+		return 0, errors.Wrap(err, "error fetching ec2 cost")
+	}
+	ebsCost, err := pkgCachingPriceFetcher.getEBSCost(m.client, h, t)
```

Comment:
should probably access the global var via some mechanism to allow you to with more injection than this reference?

Reply:
We pass the client, which is where the work happens, so I think this is fine.

Reply:
I think you should either: make this just another type and drop the interface fantasy *or* make a factory somewhere to produce objects of this type.

Reply:
dropped the interface